src/skadi/sample.cc

```diff
@@ -35,10 +34,44 @@ constexpr int16_t NO_DATA_HIGH = 16384;
 constexpr int16_t NO_DATA_LOW = -16384;
 constexpr size_t TILE_COUNT = 180 * 360;
 constexpr int8_t UNPACKED_TILES_COUNT = 50;
+const constexpr char* kDataPathPattern = "{DataPath}";
 
-// macro is faster than inline function for this..
+// macro is faster than inline function for this...
 #define out_of_range(v) v > NO_DATA_HIGH || v < NO_DATA_LOW
 
+template <typename T>
+void parallel_call(const std::function<double(const T&)>& func,
+                   std::vector<T> st,
+                   std::vector<double>& result,
+                   std::uint32_t num_threads = 1) {
+  if (!func || st.empty())
+    return;
+
+  std::uint32_t size = std::max(1U, std::min(std::thread::hardware_concurrency(), num_threads));
```

Comment:
Why limit it to `hardware_concurrency`? If the work being split is IO bound like network requests, you can definitely run a lot more threads than number of cores.

Reply:
original idea was to avoid the overhead of too many threads as this could potentially lead to performance decrease due to oversubscription. but i understood that if to play with the number of threads we could have more threads and achieve better performance.
anyway this function "parallel_call" was removed.